### PR TITLE
b/WEP-178 chech ImageTeaserBlock if the image prop exist and has value, or set placeholder image instead

### PR DESCRIPTION
**Pull request by bishoymelek**
Image Teaser Component is expecting the image to has a value but it's not the case if you add an article that has no featured image; So it broke.

### CODE REVIEW

File: examples/website/src/shared/teasers/imageTeaser.tsx [hunk 1]
```diff
@@ -125,16 +125,20 @@ export function ImageTeaser({
...
         <div className={css(ImageTeaserImageStyle)}>
           <Image
-            src={image.format === 'gif' ? image.url : image.largeURL}
-            alt={image.description || image.caption}
+            src={
+              image?.format === 'gif'
+                ? image?.url
+                : image?.largeURL || 'https://via.placeholder.com/240x240'
+            }
+            alt={image?.description || image?.caption}
```
Comment (faebebin) [inline examples/website/src/shared/teasers/imageTeaser.tsx:141]: Good workaround. Only, to me looks like there is something wrong higher up. The image prop seems not optional and `ImageData` interface requires `url` and `format` to be set. So should not the workaround be set somewhere else?

**Review by faebebin — CHANGES_REQUESTED** [review]

File: examples/website/src/shared/teasers/imageTeaser.tsx (hunk 1 above)
Comment (roosnic1) [reply examples/website/src/shared/teasers/imageTeaser.tsx:141]: @faebebin you're right. @bishoymelek will check in the api and editor if the user has selected this teaser style without an image and then throw an error.

@bishoymelek: You might want to convert this PR to a draft until you're ready again.

**Review by roosnic1 — CHANGES_REQUESTED** [review]
as discussed

File: examples/website/src/shared/teasers/imageTeaser.tsx (hunk 1 above)
Comment (bishoymelek) [reply examples/website/src/shared/teasers/imageTeaser.tsx:141]: @faebebin  Yes Absolutely correct. discussed it with @roosnic1  and now has validation in the editor and should be on the API level as well afterwords.

File: packages/editor/src/client/locales/en.json [hunk 2]
```diff
@@ -118,7 +118,8 @@
         "hideAuthors": "Hide Authors",
         "socialMediaImage": "Image for Social Media",
         "postImage": "Image for Post",
-        "metadataInfo": "Information from these fields are being used for previews in Social Media Posts. If left blank, information from general-tab will be used instead."
+        "metadataInfo": "Information from these fields are being used for previews in Social Media Posts. If left blank, information from general-tab will be used instead.",
+        "lightStyledWithNoImageError": "Light styled teaser must has an image"
```
Comment (roosnic1) [inline packages/editor/src/client/locales/en.json:122]: "Light styled teasers must have an image".

Also the location of the translation is wrong. You're in the `article` translations and i think this should go to the `pages` translations.

File: packages/editor/src/client/panel/teaserEditPanel.tsx [hunk 3]
```diff
@@ -36,7 +37,7 @@ export function TeaserEditPanel({
   onClose,
   onConfirm,
   closeLabel = 'Close'
-}: TeaserEditPanelProps) {
+}: TeaserEditPanelProps): JSX.Element {
```
Comment (roosnic1) [inline packages/editor/src/client/panel/teaserEditPanel.tsx:40]: Why did you add this?

**Review by roosnic1 — CHANGES_REQUESTED** [review]
looks good. Just some minor changes.

File: packages/editor/src/client/panel/teaserEditPanel.tsx (hunk 3 above)
Comment (faebebin) [reply packages/editor/src/client/panel/teaserEditPanel.tsx:40]: @roosnic1 FYI our linting sometimes (maybe it is just my vscode) requires return type and highlights as warning. In another project we add this return type to every FC to satisfy this constraint. @bishoymelek I think we do not require return type for FC here. But maybe we have to either fix lint rules somewhere or strictly request to have consistency within the repo and the linting.

Comment (roosnic1) [reply packages/editor/src/client/panel/teaserEditPanel.tsx:40]: @faebebin and @bishoymelek you might have to configure your editors to read the eslint config in the `package.json`

Comment (faebebin) [reply packages/editor/src/client/panel/teaserEditPanel.tsx:40]: @roosnic1 yes if you never saw this then will check my config.

File: examples/website/src/shared/teasers/imageTeaser.tsx (hunk 1 above)
Comment (faebebin) [reply examples/website/src/shared/teasers/imageTeaser.tsx:141]: @roosnic1 I checked my local settings. I have eslint activated, but that is project setting,no? Just, I did not find some setting to change by now. Don't you have those highlightings requiring return type:
image

File: packages/editor/src/client/panel/teaserEditPanel.tsx (hunk 3 above)
Comment (faebebin) [reply packages/editor/src/client/panel/teaserEditPanel.tsx:40]: @roosnic1 @roosnic1 I checked my local settings. I have eslint activated, but that is project setting,no? Just, I did not find some setting to change by now. Don't you have those highlightings requiring return type:
image

Comment (faebebin) [reply packages/editor/src/client/panel/teaserEditPanel.tsx:40]: And when runnging: yarn lint:editor

image

Comment (faebebin) [reply packages/editor/src/client/panel/teaserEditPanel.tsx:40]: ? If this is on me I will keep on working on my local settings :)

File: packages/editor/src/client/locales/en.json (hunk 2 above)
Comment (bishoymelek) [reply packages/editor/src/client/locales/en.json:122]: I guess in both article and page?

File: packages/editor/src/client/panel/teaserEditPanel.tsx (hunk 3 above)
Comment (bishoymelek) [reply packages/editor/src/client/panel/teaserEditPanel.tsx:40]: @roosnic1 @faebebin Yes I guess it's the project's eslint configuration that always makes these warnings, I actually thought to run linting before git commit so it's always aligned correctly.

File: packages/editor/src/client/locales/en.json (hunk 2 above)
Comment (bishoymelek) [reply packages/editor/src/client/locales/en.json]: @roosnic1 I added it in both article and page panels, tell me what you think

**Review by roosnic1 — APPROVED** [review]
🎉

**Review by faebebin — APPROVED** [review]
